Parsing OSV API query result fails with Vaadin 8.27.4

It looks like the data received do not match the excepted enum type we have

```
2024-12-25 16:52:41 [pool-4-thread-1] ERROR c.v.a.v.s.AppSecServiceInitListener - Error scanning vulnerabilities.
java.util.concurrent.CompletionException: com.vaadin.appsec.backend.AppSecException: Failed to read OSV API query response
        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
        at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1770)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)    
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: com.vaadin.appsec.backend.AppSecException: Failed to read OSV API query response    
        at com.vaadin.appsec.backend.OpenSourceVulnerabilityClient.queryVulnerability(OpenSourceVulnerabilityClient.java:102)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)  
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)  
        at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:992)
        at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
        at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)  
        at java.base/java.util.stream.ReferencePipeline$15$1.accept(ReferencePipeline.java:541)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)  
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)  
        at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:992)
        at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
        at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)  
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)  
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)  
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)  
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)      
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)      
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)    
        at com.vaadin.appsec.backend.OpenSourceVulnerabilityService.getVulnerabilities(OpenSourceVulnerabilityService.java:67)
        at com.vaadin.appsec.backend.VulnerabilityStore.refresh(VulnerabilityStore.java:76)    
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
        ... 6 common frames omitted
Caused by: com.fasterxml.jackson.databind.exc.InvalidFormatException: Cannot deserialize value of type `com.vaadin.appsec.backend.model.osv.response.Severity$Type` from String "CVSS_V4": not one of the values accepted for Enum class: [CVSS_V2, CVSS_V3]
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 1979] (through reference chain: com.vaadin.appsec.backend.model.osv.response.OpenSourceVulnerability["severity"]->java.util.ArrayList[1]->com.vaadin.appsec.backend.model.osv.response.Severity["type"])
        at com.fasterxml.jackson.databind.exc.InvalidFormatException.from(InvalidFormatException.java:67)
        at com.fasterxml.jackson.databind.DeserializationContext.weirdStringException(DeserializationContext.java:1958)
        at com.fasterxml.jackson.databind.DeserializationContext.handleWeirdStringValue(DeserializationContext.java:1245)
        at com.fasterxml.jackson.databind.deser.std.EnumDeserializer._deserializeAltString(EnumDeserializer.java:440)
        at com.fasterxml.jackson.databind.deser.std.EnumDeserializer._fromString(EnumDeserializer.java:304)
        at com.fasterxml.jackson.databind.deser.std.EnumDeserializer.deserialize(EnumDeserializer.java:273)
        at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:310)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177)
        at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer._deserializeFromArray(CollectionDeserializer.java:359)
        at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:244)
        at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:28)
        at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:310)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177)
        at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:342)
        at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4899)
        at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3883)        
        at com.vaadin.appsec.backend.OpenSourceVulnerabilityClient.queryVulnerability(OpenSourceVulnerabilityClient.java:100)
        ... 29 common frames omitted
```

--- a/appsec-kit-backend/pom.xml
+++ b/appsec-kit-backend/pom.xml
@@ -55,6 +55,7 @@
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
             <version>2.0.13</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>junit</groupId>

--- a/appsec-kit-backend/pom.xml
+++ b/appsec-kit-backend/pom.xml
@@ -17,7 +17,7 @@
         <dependency>
             <groupId>org.cyclonedx</groupId>
             <artifactId>cyclonedx-core-java</artifactId>
-            <version>7.3.2</version>
+            <version>9.0.4</version>
         </dependency>
         <dependency>
             <groupId>com.fasterxml.jackson.core</groupId>
@@ -34,7 +34,7 @@
         <dependency>
             <groupId>us.springett</groupId>
             <artifactId>cvss-calculator</artifactId>
-            <version>1.4.1</version>
+            <version>1.4.3</version>
         </dependency>
         <dependency>
             <groupId>org.apache.maven</groupId>
@@ -50,6 +50,11 @@
             <groupId>javax.validation</groupId>
             <artifactId>validation-api</artifactId>
             <version>2.0.1.Final</version>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <version>2.0.13</version>
         </dependency>
         <dependency>
             <groupId>junit</groupId>
@@ -73,6 +78,12 @@
             <groupId>org.mock-server</groupId>
             <artifactId>mockserver-junit-rule-no-dependencies</artifactId>
             <version>5.15.0</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.networknt</groupId>
+            <artifactId>json-schema-validator</artifactId>
+            <version>1.5.0</version>
             <scope>test</scope>
         </dependency>
     </dependencies>

--- a/appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/AppSecDTOProvider.java
+++ b/appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/AppSecDTOProvider.java
@@ -330,7 +330,7 @@ class AppSecDTOProvider {
         return vulnerability.getSeverity().stream()
                 .filter(severity -> isSupportedCvssType(severity.getType()))
                 .map(severity -> Cvss.fromVector(severity.getScore()))
-                .filter(Objects::nonNull);
+                .filter(Objects::nonNull)
                 .map(cvss -> cvss.calculateScore().getBaseScore())
                 .max(Comparator.naturalOrder()).orElse(0.0);
     }

--- a/appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/AppSecDTOProvider.java
+++ b/appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/AppSecDTOProvider.java
@@ -328,6 +328,7 @@ class AppSecDTOProvider {
     private Double getHighestCvssScoreNumber(
             OpenSourceVulnerability vulnerability) {
         return vulnerability.getSeverity().stream()
+                .filter(severity -> isSupportedCvssType(severity.getType()))
                 .map(severity -> Cvss.fromVector(severity.getScore()))
                 .map(cvss -> cvss.calculateScore().getBaseScore())
                 .max(Comparator.naturalOrder()).orElse(0.0);
@@ -343,16 +344,24 @@ class AppSecDTOProvider {
         String cvssString = "";
         double tempBaseScore = 0.0;
         for (Severity severity : vulnerability.getSeverity()) {
-            double baseScore = Cvss.fromVector(severity.getScore())
-                    .calculateScore().getBaseScore();
-            if (baseScore > tempBaseScore) {
-                tempBaseScore = baseScore;
-                cvssString = severity.getScore();
+            if (isSupportedCvssType(severity.getType())) {
+                Cvss cvss = Cvss.fromVector(severity.getScore());
+                if (cvss != null) {
+                    double baseScore = cvss.calculateScore().getBaseScore();
+                    if (baseScore > tempBaseScore) {
+                        tempBaseScore = baseScore;
+                        cvssString = severity.getScore();
+                    }
+                }
             }
         }
 
         return tempBaseScore >= highestScoreNumber ? cvssString
                 : highestScoreString;
+    }
+
+    private boolean isSupportedCvssType(Severity.Type type) {
+        return type == Severity.Type.CVSS_V2 || type == Severity.Type.CVSS_V3;
     }
 
     private Optional<String> getPatchedVersion(Affected affected) {

--- a/appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/AppSecDTOProvider.java
+++ b/appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/AppSecDTOProvider.java
@@ -330,6 +330,7 @@ class AppSecDTOProvider {
         return vulnerability.getSeverity().stream()
                 .filter(severity -> isSupportedCvssType(severity.getType()))
                 .map(severity -> Cvss.fromVector(severity.getScore()))
+                .filter(Objects::nonNull);
                 .map(cvss -> cvss.calculateScore().getBaseScore())
                 .max(Comparator.naturalOrder()).orElse(0.0);
     }

--- a/appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/model/AppSecData.java
+++ b/appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/model/AppSecData.java
@@ -24,9 +24,11 @@ public class AppSecData implements Serializable {
      * The status of the analysis for a single vulnerability.
      */
     public enum VulnerabilityStatus {
-        NOT_SET("Not set"), NOT_AFFECTED("Not affected"), FALSE_POSITIVE(
-                "False positive"), IN_TRIAGE(
-                        "In triage"), EXPLOITABLE("Exploitable");
+        NOT_SET("Not set"),
+        NOT_AFFECTED("Not affected"),
+        FALSE_POSITIVE("False positive"),
+        IN_TRIAGE("In triage"),
+        EXPLOITABLE("Exploitable");
 
         private String caption;
 

--- a/appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/model/osv/response/Severity.java
+++ b/appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/model/osv/response/Severity.java
@@ -86,7 +86,7 @@ public class Severity {
      * The severity type.
      */
     public enum Type {
-        CVSS_V2("CVSS_V2"), CVSS_V3("CVSS_V3");
+        CVSS_V2("CVSS_V2"), CVSS_V3("CVSS_V3"), CVSS_V4("CVSS_V4");
 
         private final String value;
 

--- a/appsec-kit-backend/src/test/java/com/vaadin/appsec/backend/AppSecDTOProviderTest.java
+++ b/appsec-kit-backend/src/test/java/com/vaadin/appsec/backend/AppSecDTOProviderTest.java
@@ -29,6 +29,7 @@ import com.vaadin.appsec.backend.model.osv.response.OpenSourceVulnerability;
 import com.vaadin.appsec.backend.model.osv.response.Package;
 import com.vaadin.appsec.backend.model.osv.response.Range;
 import com.vaadin.appsec.backend.model.osv.response.Reference;
+import com.vaadin.appsec.backend.model.osv.response.Severity;
 
 import static org.mockito.ArgumentMatchers.anyList;
 import static org.mockito.Mockito.mock;
@@ -88,7 +89,7 @@ public class AppSecDTOProviderTest {
 
         List<Vulnerability> vulnerabilities = dtoProvider.getVulnerabilities();
 
-        Assert.assertEquals("Mismatch in expected vulnerability count.", 1,
+        Assert.assertEquals("Mismatch in expected vulnerability count.", 3,
                 vulnerabilities.size());
     }
 
@@ -121,20 +122,47 @@ public class AppSecDTOProviderTest {
         Reference reference = new Reference(Reference.Type.WEB,
                 new URI("https://wwww.reference.com"));
 
-        Affected affected = createAffected("Maven", "org.yaml:snakeyaml",
+        Affected affected1 = createAffected("Maven", "org.yaml:snakeyaml",
                 Arrays.asList("1.32", "1.33", "1.4"), Range.Type.ECOSYSTEM,
-                new HashMap<String, String>() {
+                new HashMap<>() {
                     {
                         put("introduced", "0");
                         put("fixed", "2.0");
                     }
                 });
+        Affected affected2 = createAffected("npm", "pac-resolver", null,
+                Range.Type.SEMVER, new HashMap<>() {
+                    {
+                        put("introduced", "0");
+                        put("fixed", "5.0.0");
+                    }
+                });
+        Affected affected3 = createAffected("npm", "degenerator", null,
+                Range.Type.SEMVER, new HashMap<>() {
+                    {
+                        put("introduced", "0");
+                        put("fixed", "3.0.1");
+                    }
+                });
 
-        OpenSourceVulnerability vulnerability = createVulnerability(
+        Severity severity1 = new Severity(Severity.Type.CVSS_V3,
+                "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:N/A:N");
+        Severity severity2 = new Severity(Severity.Type.CVSS_V4,
+                "CVSS:4.0/AV:L/AC:L/AT:N/PR:L/UI:N/VC:H/VI:N/VA:N/SC:N/SI:N/SA:N");
+
+        OpenSourceVulnerability vulnerability1 = createVulnerability(
                 "GHSA-mjmj-j48q-9wg2", "CVE-2022-1471", reference,
-                Collections.singletonList(affected));
+                Collections.singletonList(affected1), Arrays.asList(serverity1, severity2));
 
-        return Collections.singletonList(vulnerability);
+        OpenSourceVulnerability vulnerability2 = createVulnerability(
+                "GHSA-9j49-mfvp-vmhm", "CVE-2021-23406", reference,
+                Collections.singletonList(affected2), Collections.emptyList());
+
+        OpenSourceVulnerability vulnerability3 = createVulnerability(
+                "GHSA-9j49-mfvp-vmhm", "CVE-2021-23406", reference,
+                Collections.singletonList(affected3), Collections.emptyList());
+
+        return Arrays.asList(vulnerability1, vulnerability2, vulnerability3);
     }
 
     private List<OpenSourceVulnerability> createVulnerabilitiesWithUnsupportedEcosystems()
@@ -160,21 +188,22 @@ public class AppSecDTOProviderTest {
 
         OpenSourceVulnerability vulnerability1 = createVulnerability(
                 "GHSA-mjmj-j48q-9wg2", "CVE-2022-1471", reference,
-                Collections.singletonList(affected1));
+                Collections.singletonList(affected1), Collections.emptyList());
         OpenSourceVulnerability vulnerability2 = createVulnerability(
                 "GHSA-9j49-mfvp-vmhm", "CVE-2021-23406", reference,
-                Collections.singletonList(affected2));
+                Collections.singletonList(affected2), Collections.emptyList());
 
         return Arrays.asList(vulnerability1, vulnerability2);
     }
 
     private OpenSourceVulnerability createVulnerability(String id, String alias,
-            Reference reference, List<Affected> affected) {
+            Reference reference, List<Affected> affected, List<Severity> severity) {
         OpenSourceVulnerability vulnerability = new OpenSourceVulnerability();
         vulnerability.setId(id);
         vulnerability.setAliases(Collections.singletonList(alias));
         vulnerability.setReferences(Collections.singletonList(reference));
         vulnerability.setAffected(affected);
+        vulnerability.setSeverity(severity);
         return vulnerability;
     }
 

--- a/appsec-kit-backend/src/test/java/com/vaadin/appsec/backend/AppSecDTOProviderTest.java
+++ b/appsec-kit-backend/src/test/java/com/vaadin/appsec/backend/AppSecDTOProviderTest.java
@@ -64,7 +64,7 @@ public class AppSecDTOProviderTest {
 
         List<Dependency> dependencies = dtoProvider.getDependencies();
 
-        Assert.assertEquals("Mismatch in expected dependency count.", 2,
+        Assert.assertEquals("Mismatch in expected dependency count.", 4,
                 dependencies.size());
     }
 
@@ -122,7 +122,7 @@ public class AppSecDTOProviderTest {
         Reference reference = new Reference(Reference.Type.WEB,
                 new URI("https://wwww.reference.com"));
 
-        Affected affected1 = createAffected("Maven", "org.yaml:snakeyaml",
+        Affected affected1 = createAffected("Maven", "org.yaml:snakeyaml1",
                 Arrays.asList("1.32", "1.33", "1.4"), Range.Type.ECOSYSTEM,
                 new HashMap<String, String>() {
                     {
@@ -130,18 +130,20 @@ public class AppSecDTOProviderTest {
                         put("fixed", "2.0");
                     }
                 });
-        Affected affected2 = createAffected("npm", "pac-resolver", null,
-                Range.Type.SEMVER, new HashMap<String, String>() {
+        Affected affected2 = createAffected("Maven", "org.yaml:snakeyaml2",
+                Arrays.asList("1.32", "1.33", "1.4"), Range.Type.ECOSYSTEM,
+                new HashMap<String, String>() {
                     {
                         put("introduced", "0");
-                        put("fixed", "5.0.0");
+                        put("fixed", "2.0");
                     }
                 });
-        Affected affected3 = createAffected("npm", "degenerator", null,
-                Range.Type.SEMVER, new HashMap<String, String>() {
+        Affected affected3 = createAffected("Maven", "org.yaml:snakeyaml3",
+                Arrays.asList("1.32", "1.33", "1.4"), Range.Type.ECOSYSTEM,
+                new HashMap<String, String>() {
                     {
                         put("introduced", "0");
-                        put("fixed", "3.0.1");
+                        put("fixed", "2.0");
                     }
                 });
 
@@ -157,11 +159,11 @@ public class AppSecDTOProviderTest {
 
         OpenSourceVulnerability vulnerability2 = createVulnerability(
                 "GHSA-9j49-mfvp-vmhm", "CVE-2021-23406", reference,
-                Collections.singletonList(affected2), Collections.emptyList());
+                Arrays.asList(affected2, affected3), Collections.emptyList());
 
         OpenSourceVulnerability vulnerability3 = createVulnerability(
                 "GHSA-9j49-mfvp-vmhm", "CVE-2021-23406", reference,
-                Collections.singletonList(affected3), Collections.emptyList());
+                Arrays.asList(affected2, affected3), Collections.emptyList());
 
         return Arrays.asList(vulnerability1, vulnerability2, vulnerability3);
     }
@@ -171,7 +173,7 @@ public class AppSecDTOProviderTest {
         Reference reference = new Reference(Reference.Type.WEB,
                 new URI("https://wwww.reference.com"));
 
-        Affected affected1 = createAffected("Maven", "org.yaml:snakeyaml",
+        Affected affected1 = createAffected("Maven", "org.yaml:snakeyaml1",
                 Arrays.asList("1.32", "1.33", "1.4"), Range.Type.ECOSYSTEM,
                 new HashMap<String, String>() {
                     {

--- a/appsec-kit-backend/src/test/java/com/vaadin/appsec/backend/AppSecDTOProviderTest.java
+++ b/appsec-kit-backend/src/test/java/com/vaadin/appsec/backend/AppSecDTOProviderTest.java
@@ -124,21 +124,21 @@ public class AppSecDTOProviderTest {
 
         Affected affected1 = createAffected("Maven", "org.yaml:snakeyaml",
                 Arrays.asList("1.32", "1.33", "1.4"), Range.Type.ECOSYSTEM,
-                new HashMap<>() {
+                new HashMap<String, String>() {
                     {
                         put("introduced", "0");
                         put("fixed", "2.0");
                     }
                 });
         Affected affected2 = createAffected("npm", "pac-resolver", null,
-                Range.Type.SEMVER, new HashMap<>() {
+                Range.Type.SEMVER, new HashMap<String, String>() {
                     {
                         put("introduced", "0");
                         put("fixed", "5.0.0");
                     }
                 });
         Affected affected3 = createAffected("npm", "degenerator", null,
-                Range.Type.SEMVER, new HashMap<>() {
+                Range.Type.SEMVER, new HashMap<String, String>() {
                     {
                         put("introduced", "0");
                         put("fixed", "3.0.1");
@@ -152,7 +152,8 @@ public class AppSecDTOProviderTest {
 
         OpenSourceVulnerability vulnerability1 = createVulnerability(
                 "GHSA-mjmj-j48q-9wg2", "CVE-2022-1471", reference,
-                Collections.singletonList(affected1), Arrays.asList(serverity1, severity2));
+                Collections.singletonList(affected1),
+                Arrays.asList(severity1, severity2));
 
         OpenSourceVulnerability vulnerability2 = createVulnerability(
                 "GHSA-9j49-mfvp-vmhm", "CVE-2021-23406", reference,
@@ -197,7 +198,8 @@ public class AppSecDTOProviderTest {
     }
 
     private OpenSourceVulnerability createVulnerability(String id, String alias,
-            Reference reference, List<Affected> affected, List<Severity> severity) {
+            Reference reference, List<Affected> affected,
+            List<Severity> severity) {
         OpenSourceVulnerability vulnerability = new OpenSourceVulnerability();
         vulnerability.setId(id);
         vulnerability.setAliases(Collections.singletonList(alias));

--- a/appsec-kit-backend/src/test/java/com/vaadin/appsec/backend/BillOfMaterialsStoreTest.java
+++ b/appsec-kit-backend/src/test/java/com/vaadin/appsec/backend/BillOfMaterialsStoreTest.java
@@ -25,6 +25,6 @@ public class BillOfMaterialsStoreTest {
                 .getResource(TEST_RESOURCE_BOM_PATH).toURI()));
 
         int components = store.getBom().getComponents().size();
-        assertEquals(2, components);
+        assertEquals(4, components);
     }
 }

--- a/appsec-kit-backend/src/test/resources/bom.json
+++ b/appsec-kit-backend/src/test/resources/bom.json
@@ -156,7 +156,7 @@
     },
     {
       "group" : "org.yaml",
-      "name" : "snakeyaml",
+      "name" : "snakeyaml1",
       "version" : "1.33",
       "description" : "YAML 1.1 parser and emitter for Java",
       "hashes" : [
@@ -200,7 +200,7 @@
           }
         }
       ],
-      "purl" : "pkg:maven/org.yaml/snakeyaml@1.33?type=jar",
+      "purl" : "pkg:maven/org.yaml/snakeyaml1@1.33?type=jar",
       "externalReferences" : [
         {
           "type" : "website",
@@ -220,18 +220,164 @@
         }
       ],
       "type" : "library",
-      "bom-ref" : "pkg:maven/org.yaml/snakeyaml@1.33?type=jar"
+      "bom-ref" : "pkg:maven/org.yaml/snakeyaml1@1.33?type=jar"
+    },
+    {
+      "group" : "org.yaml",
+      "name" : "snakeyaml2",
+      "version" : "1.33",
+      "description" : "YAML 1.1 parser and emitter for Java",
+      "hashes" : [
+        {
+          "alg" : "MD5",
+          "content" : "e0164a637c691c8cf01d29f90a709c02"
+        },
+        {
+          "alg" : "SHA-1",
+          "content" : "2cd0a87ff7df953f810c344bdf2fe3340b954c69"
+        },
+        {
+          "alg" : "SHA-256",
+          "content" : "11ff459788f0a2d781f56a4a86d7e69202cebacd0273d5269c4ae9f02f3fd8f0"
+        },
+        {
+          "alg" : "SHA-512",
+          "content" : "787b72aebb685689ab379ed8f238fc2f4175bb82d285909e39e7ff1ee1b9e64ad8878f4685dadde274f950b0dc77dca3082a52921c11094aa2a535672ae4f33c"
+        },
+        {
+          "alg" : "SHA-384",
+          "content" : "9f0a0a35df2b339830b9e78294634ea894e7096e01709c82d965e24a9fa9ac6ee15edbbbf23368dcd57741de5bf23086"
+        },
+        {
+          "alg" : "SHA3-384",
+          "content" : "91d8fd581d60be4090d3b522d8c119c12449cff2ce6c436d0e53977e1c9fea8686b2b360fb15f4c3507e4f243d7d6170"
+        },
+        {
+          "alg" : "SHA3-256",
+          "content" : "7dcf0429ecb923a66e00ec633bafeb297e3b90e1cc2be0f85443fd61367f355a"
+        },
+        {
+          "alg" : "SHA3-512",
+          "content" : "59b32910dddf63475893f7c830d58ca320cd6183704d80b8111cee7bb1674f97f5699c8ac52f905a5b9a20128b9e2b6a26a07490f217033c858d9c358bedcafc"
+        }
+      ],
+      "licenses" : [
+        {
+          "license" : {
+            "id" : "Apache-2.0"
+          }
+        }
+      ],
+      "purl" : "pkg:maven/org.yaml/snakeyaml2@1.33?type=jar",
+      "externalReferences" : [
+        {
+          "type" : "website",
+          "url" : "https://bitbucket.org/snakeyaml/snakeyaml"
+        },
+        {
+          "type" : "distribution",
+          "url" : "https://oss.sonatype.org/service/local/staging/deploy/maven2/"
+        },
+        {
+          "type" : "issue-tracker",
+          "url" : "https://bitbucket.org/snakeyaml/snakeyaml/issues"
+        },
+        {
+          "type" : "vcs",
+          "url" : "https://bitbucket.org/snakeyaml/snakeyaml/src"
+        }
+      ],
+      "type" : "library",
+      "bom-ref" : "pkg:maven/org.yaml/snakeyaml2@1.33?type=jar"
+    },
+    {
+      "group" : "org.yaml",
+      "name" : "snakeyaml3",
+      "version" : "1.33",
+      "description" : "YAML 1.1 parser and emitter for Java",
+      "hashes" : [
+        {
+          "alg" : "MD5",
+          "content" : "e0164a637c691c8cf01d29f90a709c02"
+        },
+        {
+          "alg" : "SHA-1",
+          "content" : "2cd0a87ff7df953f810c344bdf2fe3340b954c69"
+        },
+        {
+          "alg" : "SHA-256",
+          "content" : "11ff459788f0a2d781f56a4a86d7e69202cebacd0273d5269c4ae9f02f3fd8f0"
+        },
+        {
+          "alg" : "SHA-512",
+          "content" : "787b72aebb685689ab379ed8f238fc2f4175bb82d285909e39e7ff1ee1b9e64ad8878f4685dadde274f950b0dc77dca3082a52921c11094aa2a535672ae4f33c"
+        },
+        {
+          "alg" : "SHA-384",
+          "content" : "9f0a0a35df2b339830b9e78294634ea894e7096e01709c82d965e24a9fa9ac6ee15edbbbf23368dcd57741de5bf23086"
+        },
+        {
+          "alg" : "SHA3-384",
+          "content" : "91d8fd581d60be4090d3b522d8c119c12449cff2ce6c436d0e53977e1c9fea8686b2b360fb15f4c3507e4f243d7d6170"
+        },
+        {
+          "alg" : "SHA3-256",
+          "content" : "7dcf0429ecb923a66e00ec633bafeb297e3b90e1cc2be0f85443fd61367f355a"
+        },
+        {
+          "alg" : "SHA3-512",
+          "content" : "59b32910dddf63475893f7c830d58ca320cd6183704d80b8111cee7bb1674f97f5699c8ac52f905a5b9a20128b9e2b6a26a07490f217033c858d9c358bedcafc"
+        }
+      ],
+      "licenses" : [
+        {
+          "license" : {
+            "id" : "Apache-2.0"
+          }
+        }
+      ],
+      "purl" : "pkg:maven/org.yaml/snakeyaml3@1.33?type=jar",
+      "externalReferences" : [
+        {
+          "type" : "website",
+          "url" : "https://bitbucket.org/snakeyaml/snakeyaml"
+        },
+        {
+          "type" : "distribution",
+          "url" : "https://oss.sonatype.org/service/local/staging/deploy/maven2/"
+        },
+        {
+          "type" : "issue-tracker",
+          "url" : "https://bitbucket.org/snakeyaml/snakeyaml/issues"
+        },
+        {
+          "type" : "vcs",
+          "url" : "https://bitbucket.org/snakeyaml/snakeyaml/src"
+        }
+      ],
+      "type" : "library",
+      "bom-ref" : "pkg:maven/org.yaml/snakeyaml3@1.33?type=jar"
     }
   ],
   "dependencies": [
     {
       "ref" : "pkg:maven/org.springframework.boot/spring-boot-starter@3.1.4?type=jar",
       "dependsOn" : [
-        "pkg:maven/org.yaml/snakeyaml@1.33?type=jar"
+        "pkg:maven/org.yaml/snakeyaml1@1.33?type=jar",
+        "pkg:maven/org.yaml/snakeyaml2@1.33?type=jar",
+        "pkg:maven/org.yaml/snakeyaml3@1.33?type=jar"
       ]
     },
     {
-      "ref" : "pkg:maven/org.yaml/snakeyaml@1.33?type=jar",
+      "ref" : "pkg:maven/org.yaml/snakeyaml1@1.33?type=jar",
+      "dependsOn" : [ ]
+    },
+    {
+      "ref" : "pkg:maven/org.yaml/snakeyaml2@1.33?type=jar",
+      "dependsOn" : [ ]
+    },
+    {
+      "ref" : "pkg:maven/org.yaml/snakeyam3@1.33?type=jar",
       "dependsOn" : [ ]
     }
   ]

